sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/FastestServerFetcher.kt

```diff
@@ -120,25 +121,33 @@ internal class FastestServerFetcher(
             LightWalletClient.new(
                 context = context,
                 lightWalletEndpoint = endpoint,
-                singleRequestTimeout = 5.seconds
+                singleRequestTimeout = FETCH_THRESHOLD
             )
 
-        val remoteInfo: LightWalletEndpointInfoUnsafe
+        val remoteInfo: LightWalletEndpointInfoUnsafe?
         val getServerInfoDuration =
             measureTime {
                 remoteInfo =
-                    when (val response = lightWalletClient.getServerInfo()) {
-                        is Response.Success -> response.result
-                        is Response.Failure -> {
-                            logRuledOut("getServerInfo failed", response.toThrowable())
-                            return null
+                    withTimeoutOrNull(5.seconds) { // 5 seconds timeout in case server is very unresponsive
+                        when (val response = lightWalletClient.getServerInfo()) {
+                            is Response.Success -> response.result
+                            is Response.Failure -> {
+                                logRuledOut("getServerInfo failed", response.toThrowable())
+                                null
+                            }
                         }
                     }
             }
 
+        if (remoteInfo == null) {
+            lightWalletClient.close()
+            return null
+        }
+
         // Check network type
         if (!remoteInfo.matchingNetwork(network.networkName)) {
             logRuledOut("matchingNetwork failed")
+            lightWalletClient.close()
```

Comment:
Could we improve these dedicated `close()`  calls by using `.use{}`?

Reply:
actually we cannot, because some we need to close some not :) 'use' would close them all